extensions/2.0/Khronos/KHR_lights_rect_area/README.md

```diff
@@ -0,0 +1,142 @@
+
+## Status
+
+Proposed
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension defines a rectangular area light for use with glTF 2.0.
+
+Many 3D tools and engines support built-in implementations of rect area types. Using this extension, tools can export and engines can import these lights. 
+
+This extension defines a single "rect" area light type: `rect`.
+
+These lights are referenced by nodes and inherit the transform of that node.
+
+A conforming implementation of this extension must be able to load light data defined in the asset and has to render the asset using those lights. 
+
+## Defining Rect Area Lights
+
+Lights are defined within a dictionary property in the glTF scene description file, by adding an `extensions` property to the top-level glTF 2.0 object and defining a `KHR_lights_rect_area` property with a `lights` array inside it.
+
+The area light is assumed to point down the -Z axis.
+
+`width` is along the X-axis, and assumed to be centered.
+
+`height` is defined along the y-axis, and assumed to be centered.
+
+The following example defines a white-colored rect area light.
+
+```javascript
+"extensions": {
+    "KHR_lights_rect_area" : {
+        "lights": [
+            {
+                "width": 1.0,
+                "height": 1.0,
+                "intensity": 1.0,
+                "color": [
+                    1.0,
+                    1.0,
+                    1.0
+                ],
+            }
+        ]
+    }
+}
+```
+
+## Adding Light Instances to Nodes
+
+Lights must be attached to a node by defining the `extensions.KHR_lights_rect_area` property and, within that, an index into the `lights` array using the `light` property.
+
+```javascript
+"nodes" : [
+    {
+        "extensions" : {
+            "KHR_lights_rect_area" : {
+                "light" : 0
+            }
+        }
+    }            
+]
+```
+
+The light will inherit the transform of the node.
+
+## Rect Area Properties
+
+| Property | Description | Required |
+|:-----------------------|:------------------------------------------| :--------------------------|
+| `name` | Name of the light. | No, Default: `""` |
+| `color` | RGB value for light's color in linear space. | No, Default: `[1.0, 1.0, 1.0]` |
+| `intensity` | Brightness of light in. The units that this is defined in depend on the type of light in terms of luminous flux per unit area (1 lux = 1 lumen/m^2 = 1 candela * sr / m^2) | No, Default: `1.0` |
+| `width` | The centered width of the light along the X-axis. | :white_check_mark: Yes |
+| `height` | The centered height of the light along the Y-axis. | :white_check_mark: Yes |
+
+QUESTION: If an area light is scaled down by parent transforms, does it emit less light because it has less world space area?  Currently this would be the behavior given the current definition.  This seems intuitively like it makes sense.
```

Comment:
What do you think about the simplifying the object a bit, specifically:
- define an area light source as 1x1 square, set intensity relative to 1m<sup>2</sup>;
- remove `width` and `height` properties;
- derive real-world size from node transforms?

Reply:
You are current that one could do this, but in my experience every implementation I've see has width and height as explicit parameters:

https://docs.unrealengine.com/en-US/BuildingWorlds/LightingAndShadows/LightTypes/RectLights/index.html

http://playcanvas.github.io/#graphics/area-lights.html

https://docs.chaosgroup.com/display/VRAYNUKE/Rectangle+or+Area+Light+%7C+VRayLightRect

Reply:
Okay, I see. My concern here is that this design makes it possible to represent the same light in infinite amount of ways by combining light's own dimensions with node transforms. Such flexibility should be thoroughly evaluated for a portable runtime transmission format.

On the other hand, if we can precisely specify how all these properties interact with each other, it might be better to keep them.

Reply:
I implemented it using a naming convention and I went for 1x1 + node transform.